Make documentation more precise about how rules are enabled (#3889)
detekt/detekt: docs/pages/extensions.md

}
```

If you want your rule to be configurable, write down your properties inside the detekt.yml file.

> Please note that this will only take effect, if the Config object is passed on by the RuleSetProvider

Reviewer: ```suggestion
Please note that this will only take effect, if the `Config` object is passed on by the `RuleSetProvider`
```

> the the rule itself.

Reviewer: ```suggestion
to the rule itself.
```

##### Pitfalls

- All rules are disabled by default and have to be explicitly enabled in the `detekt` yaml configuration file.

> - If you do not pass the Config object from the rule set provider to the rule, the rule is active but you will not be able to use

Reviewer: ```suggestion
- If you do not pass the `Config` object from the `RuleSetProvider` to the rule, the rule is active, but you will not be able to use
```
